src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs

```diff
@@ -149,7 +144,7 @@ private static PrimarySignature CreatePrimarySignature(CmsSigner cmsSigner, Sign
 
             return PrimarySignature.Load(cms);
         }
-
+#if SUPPORTS_FULL_SIGNING
```

Comment:
The line above https://github.com/NuGet/NuGet.Client/pull/2706/files#diff-9f9f0fc95fc31c850ddf340ed90c8d57R134
`cms.ComputeSignature(cmsSigner);` needs to be changed to be `cms.ComputeSignature(cmsSigner, false);`

That parameter is to use a silent context or not, and the default is true. This means that in a silent context, a PIN prompt won't appear if the CNG/CAPI api requires one (like for a smartcard or usb token).

The .NET Framework didn't handle this correctly, and it worked. The behavior was made "correct" on .NET Core, so specifying `false` for the silent context is required.